src/clj_foundation/errors.clj

```diff
@@ -109,3 +106,65 @@ attempts, the last caught exception is re-thrown."
   The remaining parameters (if any) are that function's arguments."
   [tries pause-millis]
   (fn [f & args] (apply retry tries pause-millis f args)))
+
+
+(s/defn replace-if :- s/Any
```

Comment:
You could have a module for boundary conditions.  As for where and how long, code location choices are reframed when you extract a library.  I don't thing anyone would be looking for `value-or` in `errors.clj`